File: hackerrank/30_day_of_code/day_10_binary_numbers.js
// https://www.hackerrank.com/challenges/30-binary-numbers/problem
"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", inputStdin => {
    inputString += inputStdin;
});

process.stdin.on("end", _ => {
    inputString = inputString
        .replace(/\s*$/, "")
        .split("\n")
        .map(str => str.replace(/\s*$/, ""));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

const binMax = n =>
    n
        .toString(2)
        .split("0")
        .sort((a, b) => b.length - a.length)[0].length;

function main() {
    const n = parseInt(readLine(), 10);
    console.log(binMax(n));
}

File: hackerrank/30_day_of_code/day_10_binary_numbers.py
# https://www.hackerrank.com/challenges/30-binary-numbers/problem


#!/bin/python3

import math
import os
import random
import re
import sys


def toBinary(n):
    st = bin(n)[2:]
    return len(max(st.split('0')))


if __name__ == '__main__':
    n = int(input())
    print(toBinary(n))
